### PR TITLE
Lab 3 - Letter frequency

### LINKED ISSUE #167: Lab 3 - Letter frequency
Create an executable go program that takes a string as the input and returns a mapping of each letter to its frequency.

### PR DESCRIPTION
**Pull request by kasunfdo**
Fixes #167

Review of colleague's PR #201

#### Changes proposed in this PR:
* Implement letter frequency

### CODE REVIEW

File: 03_letters/kasunfdo/main_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		"i:3", "k:1", "l:1", "n:1", "o:2", "p:1", "r:1", "s:1", "u:1", "v:1", "w:1", "x:1", "y:1", "z:1"}
+
+	actual := sortLetters(letters(testData))
+
+	if !reflect.DeepEqual(expected, actual) {
+		t.Errorf("Unexpected output in main()\nexpected: %q\nactual: %q", expected, actual)
+	}
+}
+
+func TestRandomStrWithSymbols(t *testing.T) {
```
Comment (CameronSchafer) [inline 03_letters/kasunfdo/main_test.go]: Empty test function. Missing code or unused test case?

File: 03_letters/kasunfdo/main_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+func TestSortLetters(t *testing.T) {
+	type test struct {
+		input    string
+		expected []string
+	}
+
+	tests := []test{
+		{input: "Two Driven Jocks Help Fax My Big Quiz!",
+			expected: []string{			" :7", "!:1", "B:1", "D:1", "F:1", "H:1", "J:1", "M:1", "Q:1", "T:1", "a:1", "c:1", "e:2", "g:1",
```
Comment (golangcibot) [inline 03_letters/kasunfdo/main_test.go]: File is not `gofmt`-ed with `-s` (from `gofmt`)

File: 03_letters/kasunfdo/main_test.go (hunk 1 above)
Comment (kasunfdo) [reply 03_letters/kasunfdo/main_test.go]: Thanks @CameronSchafer. I fixed that

**Review by juliaogris — CHANGES_REQUESTED** [review]
Again -  well done, looks pretty complete!

Please add link to PR review of colleague to your PR description (we do this to help our load).
Test some corner cases - empty string, 2, 3, 4 rune string, white space on either side.

File: 03_letters/kasunfdo/main.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+	for _, char := range s {
+		charMap[char]++
+	}
+
+	return charMap
+}
+
+func sortLetters(m map[rune]int) []string {
+	out := make([]string, len(m))
```
Comment (juliaogris) [inline 03_letters/kasunfdo/main.go:24]: Nit: I like my variables declared as close as possible to there usage (less jumping around when reading) so i'd put `out := ...` just above the second loop.

**Review by juliaogris — APPROVED** [review]
Great work again!

I've only got a couple of nitpicks left, you can keep them for the next PR ;)
- imo you are a bit excessive with whitespace (blanklines) at least within functions in `main.go` - but I guess we can call this a matter of taste
- one commit message starts with `Refactored` should be `Refactor`
- imo it's best to declare variables as local as possible to their use and not all at the beginning of the function - to me this reads easier (less jumping around/ context switching)

File: 03_letters/kasunfdo/main.go (hunk 3 above)
Comment (kasunfdo) [reply 03_letters/kasunfdo/main.go:24]: I agree. Thanks Julia! :)
